### PR TITLE
Release/v0.6.0 refactor

### PR DESCRIPTION
**Pull request by yaniaiciin2**
> Now ApplicationRunner methods are private

> Changed tests in order to cover the entire Runner code

### CODE REVIEW

File: src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java [hunk 1]
```diff
@@ -84,17 +83,21 @@ static void setDynamicProperties(DynamicPropertyRegistry registry) {
     @Test
     void shouldSaveAuditRecordRoot() throws JsonProcessingException, NoSuchAlgorithmException {
         // Arrange
-        String expectedAuditRecordRootHash = "bfa2600854f1902943af3094a43bbf28188bf88fa2280c2d8ed1bb4f4524d2dc";
+        auditRecordRepository.deleteAll().block();
```
Comment (oriolcanades) [inline src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java:86]: This is not a solution.

File: src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java [hunk 2]
```diff
@@ -84,17 +83,21 @@ static void setDynamicProperties(DynamicPropertyRegistry registry) {
...
+        String expectedAuditRecordRootHash = "f291c0096b7c3e10a52db72ada76676ed2a928b7fd9e91ab9f1ccb7614d8bd08";
         // Calculate the hash of the AuditRecordRoot and set them.
         // The hashLink is the hash of the AuditRecordRoot because it is the first record
         String auditRecordRootHash = calculateSHA256(objectMapper.writeValueAsString(auditRecordRoot));
+        System.out.println("auditRecordRoot:" + objectMapper.writeValueAsString(auditRecordRoot));
```
Comment (oriolcanades) [inline src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java:91]: To delete — debug purposes

File: src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java [hunk 3]
```diff
@@ -84,17 +83,21 @@ static void setDynamicProperties(DynamicPropertyRegistry registry) {
...
         auditRecordRoot.setHash(auditRecordRootHash);
         auditRecordRoot.setHashLink(auditRecordRootHash);
         // Save AuditRecordRoot
         AuditRecord auditRecordRootMono = auditRecordRepository.save(auditRecordRoot).block();
         // Assertion for AuditRecordRoot
         assert auditRecordRootMono != null;
         // The hash of the AuditRecordRoot is the hash expected and calculated previously
+        System.out.println("auditRecordRootHash: " + auditRecordRootHash);
```
Comment (oriolcanades) [inline src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java:99]: To delete — debug purposes

File: src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java [hunk 4]
```diff
@@ -108,16 +111,17 @@ void shouldSaveAuditRecordRoot() throws JsonProcessingException, NoSuchAlgorithm
...
          * you will get the hashLink of the new AuditRecord ;)
          */
         // Arrange
-        String expectedAuditRecordHash = "c69ecda4a694be164a9be89adff48ab01b281dd88b236ffd35f5369d397bf3bc";
-        String expectedAuditRecordHashLink = "8337ddb39415f3b647d1d8756dc7f46a176ca5b35c784ce0fa3e27bafc32ce98";
+        String expectedAuditRecordHash = "8008569e61c6bea20c3d35c70f742934b8091626f844416baf2855199964dc58";
+        String expectedAuditRecordHashLink = "f09ba1b7cd0782e43e7f85ba54c81c1e9a27ebee1e51411e14b0790c025c48e0";
         // Get the most recent AuditRecord from the database
         AuditRecord auditRecordFound = auditRecordRepository.findMostRecentAuditRecord().block();
+        System.out.println("auditRecordFound: " + auditRecordFound);
```
Comment (oriolcanades) [inline src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java:124]: To delete — debug purposes

File: src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java [hunk 5]
```diff
@@ -126,6 +130,7 @@ void shouldSaveAuditRecordGuaranteeImmutability() throws JsonProcessingException
         auditRecord.setHashLink(auditRecordHashLink);
         // Save AuditRecord
         AuditRecord auditRecordMono = auditRecordRepository.save(auditRecord).block();
+        System.out.println(auditRecordMono);
```
Comment (oriolcanades) [inline src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java:133]: To delete — debug purposes

File: src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java [hunk 6]
```diff
@@ -136,6 +141,7 @@ void shouldSaveAuditRecordGuaranteeImmutability() throws JsonProcessingException
     @Test
     void shouldRetrieveAllAuditRecords() {
         Flux<AuditRecord> auditRecordFlux = auditRecordRepository.findAll();
+        System.out.println(auditRecordFlux);
```
Comment (oriolcanades) [inline src/test/java/es/in2/desmos/domain/repositories/AuditRecordRepositoryIT.java:144]: To delete — debug purposes

File: src/test/java/es/in2/desmos/bdd/PublishWorkflowBehaviorTest.java [hunk 7]
```diff
@@ -50,6 +47,11 @@ static void setDynamicProperties(DynamicPropertyRegistry registry) {
         ContainerManager.postgresqlProperties(registry);
     }
 
+    @AfterEach
```
Comment (oriolcanades) [inline src/test/java/es/in2/desmos/bdd/PublishWorkflowBehaviorTest.java:50]: Is this used?

**Review by oriolcanades — CHANGES_REQUESTED** [review]

File: src/test/java/es/in2/desmos/bdd/PublishWorkflowBehaviorTest.java (hunk 7 above)
Comment (yaniaiciin2) [reply src/test/java/es/in2/desmos/bdd/PublishWorkflowBehaviorTest.java:50]: Yes, actually this is the solution

File: src/test/java/es/in2/desmos/bdd/PublishWorkflowBehaviorTest.java [hunk 8]
```diff
@@ -9,10 +9,8 @@
 import es.in2.desmos.domain.models.BrokerNotification;
 import es.in2.desmos.domain.repositories.AuditRecordRepository;
 import es.in2.desmos.domain.services.api.QueueService;
-import org.junit.jupiter.api.MethodOrderer;
-import org.junit.jupiter.api.Order;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.Before;
```
Comment (oriolcanades) [inline src/test/java/es/in2/desmos/bdd/PublishWorkflowBehaviorTest.java:12]: If you are using AfterEach, why are you importing Before?

**Review by oriolcanades — CHANGES_REQUESTED** [review]

**Review by oriolcanades — APPROVED** [review]

**Pull request merged**